pilot/pkg/model/push_context.go

```diff
@@ -1569,6 +1577,42 @@ func (ps *PushContext) initVirtualServices(env *Environment) error {
 				}
 			}
 		}
+
+		if features.FilterGatewayClusterConfig {
```

Comment:
is it feasible to re-use virtualServiceDestinations instead of copy?

Reply:
I don't think so, because then we can't do O(1) lookup by hostname, and `util/set` does not support types other than int, float and string

Reply:
Fixed